### PR TITLE
TIP-901: Small fixes towards SF4

### PR DESCRIPTION
**Pull request by jjanvier**
Easier to read commit by commit. 

The kernel is now located in `src/Kernel.php` for Symfony4.
ACL support is removed from Symfony. We need an extra `AclBundle`.

### CODE REVIEW

File: src/Akeneo/Tool/Bundle/ElasticsearchBundle/tests/integration/UpdateIndexMappingIntegration.php [hunk 1]
```diff
@@ -16,7 +16,7 @@ class UpdateIndexMappingIntegration extends TestCase
 {
     public function test_it_moves_the_index_and_the_pqb_still_works()
     {
-        $indexHost = $this->getParameter('index_hosts');
+        $indexHost = getenv('APP_INDEX_HOSTS');
```
Comment (AymericPlanche) [inline src/Akeneo/Tool/Bundle/ElasticsearchBundle/tests/integration/UpdateIndexMappingIntegration.php]: :warning: this will not work in the future, in the new version of dotenv component the variables defined in .env are not set as real system environment variables. https://github.com/symfony/dotenv/blob/master/CHANGELOG.md#500 up to discuss about this IRL ;)

Comment (jjanvier) [reply src/Akeneo/Tool/Bundle/ElasticsearchBundle/tests/integration/UpdateIndexMappingIntegration.php]: @AymericPlanche but isn't `.env` just a "hack"? shouldn't we use real env vars, like in production? I'm quite lost about what we should or should not do about env vars...

I can revert this commit, no problem btw

Comment (anaelChardan) [reply src/Akeneo/Tool/Bundle/ElasticsearchBundle/tests/integration/UpdateIndexMappingIntegration.php]: You could set the parameter index host by fetching the env directly in the config file no?

File: app/config/config.yml [hunk 2]
```diff
@@ -52,5 +52,3 @@ akeneo_pim_structure:
 
 parameters:
     pim_ce_dev_src_folder_location: '%kernel.project_dir%'
-    index_hosts: '%env(APP_INDEX_HOSTS)%'
```
Comment (anaelChardan) [inline app/config/config.yml]: I would let it like that

File: src/Akeneo/Tool/Bundle/ElasticsearchBundle/tests/integration/UpdateIndexMappingIntegration.php (hunk 1 above)
Comment (AymericPlanche) [reply src/Akeneo/Tool/Bundle/ElasticsearchBundle/tests/integration/UpdateIndexMappingIntegration.php]: First: Why do we want to use environment variable based configuration value ?
-> Cloud deployment is based on production docker images. We don't want to generate and maintain configuration files that could be mounted as volumes in docker images when we have to deploy them in production. 
The cloud way for application configuration is environment variables everywhere, the one used in production are real / system environment variables, if you do a `export` command in a production container, you'll be able to see the environment variables.

-> Regarding the dev environment(s), as we can/have to manage multiple application instances in one laptop, it's easier to be able to have a `.env` file that will be `loaded` by the `symfony\DotEnv` component within the application instead of exporting env var each time we need to switch from a ee-3.0 and a ee-3.1 for instance.

to sum up: the fact that we manage default values via a `.env` file is great for dev/ci environment and allow to be production ready, but it does not mean that the application need to read the value directly from the `getenv()` php method, we need to have the configuration keys in configuration yaml files, the value will be dynamic using `%env(...)%` syntax.

Comment (jjanvier) [reply src/Akeneo/Tool/Bundle/ElasticsearchBundle/tests/integration/UpdateIndexMappingIntegration.php]: OK, I'll revert this commit
thanks @AymericPlanche

Comment (ahocquard) [reply src/Akeneo/Tool/Bundle/ElasticsearchBundle/tests/integration/UpdateIndexMappingIntegration.php]: > but it does not mean that the application need to read the value directly from the getenv()

In my opinion it means that the code will work in production but not in development...

Comment (AymericPlanche) [reply src/Akeneo/Tool/Bundle/ElasticsearchBundle/tests/integration/UpdateIndexMappingIntegration.php]: I think it's really important that everyone have to be convinced about this subject, maybe we could make a presentation to all the team. cc @jjanvier
